src/pyats/contrib/creators/topology.py

```diff
@@ -0,0 +1,656 @@
+import os
+from collections import OrderedDict
+import ipaddress
+import argparse
+import logging
+import sys
+import subprocess
+import re
+from yaml import (ScalarEvent, MappingEndEvent, MappingStartEvent, AliasEvent,
+                  SequenceEndEvent, SequenceStartEvent, DocumentEndEvent,
+                  DocumentStartEvent, StreamEndEvent, StreamStartEvent, emit,
+                  resolver, Dumper, dump, YAMLError, safe_load)
+from concurrent.futures import ThreadPoolExecutor
+import pprint
+from genie.conf.base import Testbed, Device, Interface, Link
+from genie.conf import Genie
+from genie.testbed import load
+from pyats.async_ import pcall
+from .creator import TestbedCreator
+from genie.metaparser.util.exceptions import SchemaEmptyParserError
+import time
+log = logging.getLogger(__name__)
+    
+os_list = ['nxos', 'iosxr', 'iosxe', 'ios']
+
+
+
+class Topology(TestbedCreator):
+    
+    """ Topology class (TestbedCreator)
+
+    Creates a yaml file that
+
+    """
+    cdp_set = set()
+    lldp_set = set()
+    visited_devices = set()
+
+    def _init_arguments(self):
+        """ Specifies the arguments for the creator.
+        Returns:
+            dict: Arguments for the creator.
+        """
+        return {
+            'required': ['testbed_name'],
+            'optional': {
+                'config_off': False,
+                'all_interfaces': False,
+                'exclude_network': '',

```

Comment:
exclude and ignore are almost same meaning here. let's use either one. and multiple networks can be also given to exclude_network, right? please add `s` end of the key.

```
exclude_networks
exclude_interfaces

Reply:
Done